components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/claims/impl/DefaultClaimHandler.java

```diff
@@ -1100,6 +1107,91 @@ private void handleRoleClaim(AuthenticationContext context, Map<String, String>
         }
     }
 
+    private void handleApplicationRolesForFederatedUser(StepConfig stepConfig, AuthenticationContext context,
+                                                        ClaimMapping[] idPClaimMappings,
+                                                        Map<String, String> remoteClaims) throws FrameworkException {
+
+        if (idPClaimMappings != null && stepConfig.isSubjectAttributeStep()) {
+            boolean hasAppRoleClaimMapping = false;
+            String idpAppRoleClaimUri = null;
+            for (ClaimMapping claimMapping : idPClaimMappings) {
+                if (claimMapping.getLocalClaim().getClaimUri().equals(FrameworkConstants.APP_ROLES_CLAIM)) {
+                    hasAppRoleClaimMapping = true;
+                    idpAppRoleClaimUri = claimMapping.getRemoteClaim().getClaimUri();
+                    break;
+                }
+            }
+            String requestedAppRoleClaim = context.getSequenceConfig().getApplicationConfig()
+                    .getRequestedClaimMappings().get(FrameworkConstants.APP_ROLES_CLAIM);
+            boolean isScopeValidated = false;
```

Comment:
This should not be handled in authentication framework

Reply:
Addressed in https://github.com/wso2/carbon-identity-framework/commit/453eaf009c1935bd93f7f855765f6f09de8cfe10